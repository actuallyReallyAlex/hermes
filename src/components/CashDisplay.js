import React from 'react'
import PropTypes from 'prop-types'
import { Heading, Text, Box } from 'grommet'
import { connect } from 'react-redux'

const CashDisplay = ({ cash }) => {
  return (
    <Box margin={{ bottom: 'small' }}>
      <Heading level="2" margin={{ bottom: 'xsmall' }}>
        Cash:
      </Heading>
      <Text>{cash}</Text>
    </Box>
  )
}

CashDisplay.propTypes = {
  cash: PropTypes.number.isRequired
}

const mapStateToProps = ({ user }) => ({ cash: user.cash })

export default connect(mapStateToProps)(CashDisplay)
