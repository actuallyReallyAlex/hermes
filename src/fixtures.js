export const defaultState = {
  ship: {
    cargo: {
      items: [],
      volumeRemaining: 5
    },
    location: {
      name: null,
      value: null
    }
  },
  ui: {
    view: 'Ship'
  },
  user: {
    cash: 100
  },
  world: {
    isTimerRunning: false,
    planets: [
      {
        id: "0",
        isHomePlanet: true,
        location: 0,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '0',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '1',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '2',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '3',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '4',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          }
        ],
        name: 'Test Planet 1'
      },
      {
        id: "1",
        isHomePlanet: false,
        location: 50,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '5',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '6',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '7',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '8',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '9',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          }
        ],
        name: 'Test Planet 2'
      },
      {
        id: "2",
        isHomePlanet: false,
        location: 100,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '10',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '11',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '12',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '13',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '14',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          }
        ],
        name: 'Test Planet 3'
      }
    ]
  }
}
