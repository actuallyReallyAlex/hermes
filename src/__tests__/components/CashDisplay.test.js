import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { defaultState, mockStore } from '../../fixtures'
import CashDisplay from '../../components/CashDisplay'

describe('<CashDisplay />', () => {
  it('Should render the <CashDisplay /> component.', () => {
    const container = render(
      <Provider store={mockStore(defaultState)}>
        <CashDisplay />
      </Provider>
    )
    expect(container.asFragment()).toMatchSnapshot()
  })
})
