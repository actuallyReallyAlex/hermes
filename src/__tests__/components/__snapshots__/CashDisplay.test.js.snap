// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CashDisplay /> Should render the <CashDisplay /> component. 1`] = `
<DocumentFragment>
  <div
    class="StyledBox-sc-13pk1d4-0 lfluzk"
  >
    <h2
      class="StyledHeading-sc-1rdh4aw-0 gBloPj"
    >
      Cash:
    </h2>
    <span
      class="StyledText-sc-1sadyjn-0 fWSbXS"
    >
      100
    </span>
  </div>
</DocumentFragment>
`;
